portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -37,6 +38,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String username = request.getParameter("username");
     String text = request.getParameter("text");
+    // if any of the necessary fields are absent, raise an error
+    if (username == null || text == null || text.trim().length() == 0) {
```

Comment:
I suggest making this into different Ifs and give corresponding error message, (invalid user name, no text etc). This makes it easier to test and we can test each condition.